tabular/setup.py

```diff
@@ -41,6 +41,7 @@ def create_version_file():
     'numpy>=1.16.0',
     'scipy>=1.3.3',
     'catboost>=0.23.0,<0.25',
+    'xgboost>=1.2,<1.3',
```

Comment:
For this PR, lets not add the hard dependency, similar to how we don't depend on FastAI despite having a FASTAI model. We will plan to add in as optional dependency in future PR for further modularization.